group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -272,6 +272,8 @@ public GroupCoordinatorShard build() {
                 .withAuthorizerPlugin(authorizerPlugin)
                 .build();
 
+            groupConfigManager.registerListener(groupMetadataManager);
```

Comment:
I don't think this will work. There is only a single groupConfigManager, but there is one group metadata manager per metrics shared. I think we have to register the group coordinator service as a listener.

Reply:
I don't think there is anything out htere that is kept up-to-date. There is only reading the code. I also did not know how many instances of `GroupConfigManager` there are, I followed the code.